src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -912,9 +910,6 @@ winrt::Windows::Foundation::IAsyncOperation<winrt::hstring> AppHost::_GetWindowL
     }
     CATCH_LOG()
 
-    // go back to give the result to the peasant.
-    co_await peasant_thread;

```

Comment:
This does worry me - we're now returning the results _on the UI[^1] thread_. I wanna make sure that the callers don't do anything too expensive on the UI thread. 

[^1]: The window's UI thread, not the _main_ main thread

Reply:
* `WindowEmperor::_saveWindowLayouts`
  * go to bg thread
  * (couple plumbing layers)
  * *bg thread* `Monarch::GetAllWindowLayouts`: collect all the window layouts for each peasant
    * *bg thread* `Peasant::GetWindowLayout` raises the `GetWindowLayoutRequested`. We stuff into the args _this async function. 
    * *bg thread* Control returns into `Peasant::GetWindowLayout` to process the async operation. We call `get`, to trigger this function here. 
    * *UI thread* `Peasant::GetWindowLayout` gets the `layoutJson` we stuffed into the args. 
    * we return that json up to the Monarch
  * pop stack a couple times
  * *UI thread* return into `WindowEmperor::_saveWindowLayouts`
    * **UI thread** `AppLogic::SaveWindowLayoutJsons` <---- This is getting called on the UI thread now! (_but that seems okay_)
      * **UI thread** parse the json into a `WindowLayout` (`WindowLayout::FromJson(json)` for each window)
      * **UI thread** calls `ApplicationState::SharedInstance().PersistedWindowLayouts`
        * triggers a throttled call to persist the state.

Okay so really, the main change here is we do the json->`WindowLayout` parsing on the UI thread now. In my experience, JSON parsing is _absolutely_ fine to do on the UI thread - we're never dealing with particularly large blobs, and jsoncpp is fairly performant. 

This honestly might make some things better. Before, each peasant would hop from the BG, to the UI, then back to the original BG thread. Then the monarch would ask the next one (to do those same hops). *Now*, I think we'll just hop from the BG to the UI thread for the first one, and when the monarch gets to the next peasant, it won't actually change thread[^1]. 

So overall, I think this should be fine


[^1]: assuming a `resume_forground` on the UI thread results in a noop